packages/app/src/cli/models/app/loader.ts

```diff
@@ -576,6 +576,47 @@ class AppConfigurationLoader {
   }
 }
 
+/**
+ * Runs a promise, racing it against a timeout to cut short execution.
+ *
+ * Any errors or promise rejections are swallowed and replaced with undefined
+ *
+ * @param getActionPromise - function returning a promise for the work you want to run
+ * @param timeoutMs - timeout in ms to race the work against
+ * @returns the result of the given action, or undefined, in at most `timeoutMs` milliseconds
+ */
+async function resolvePromiseOrUndefinedWithATimeout<T>(
+  getActionPromise: () => Promise<T>,
+  timeoutMs = 500,
+): Promise<T | undefined> {
+  const actionPromise = (async () => {
+    let actionResult
+    try {
+      actionResult = await getActionPromise()
+      // eslint-disable-next-line no-catch-all/no-catch-all
+    } catch {
+      // leave as-is
+    }
+    return actionResult
+  })()
+
+  const timeoutPromise = new Promise<T>((_resolve, reject) => {
+    const id = setTimeout(() => {
+      clearTimeout(id)
+      reject(new Error('Operation timed out'))
+    }, timeoutMs)
+  })
+
+  try {
+    const result = await Promise.race([actionPromise, timeoutPromise])
+    return result
+    // eslint-disable-next-line no-catch-all/no-catch-all
+  } catch {
+    // we'd only hit this if both promises fail, happy to return with an undefined
+    return undefined
+  }
+}
+
```

Comment:
I was hoping we could wrap the `git` helper (the one from `simple-git`) that initializes the repo, so that we can handle all git timeouts in a single place. what do you think?

Reply:
its used by hydrogen so I'm not sure I can safely change its behaviour